File: 04/ruby/factorial.rb
#! /usr/bin/ruby
# factorial.rb calculates the factorial value of integers
#
# Author: Jonathan Winkle
# 3/3/2020
#
# Input:  an integer n
# Output: the value of n!


if __FILE__ == $0
    print "Enter a value to calculate its factorial: "
    n = gets.chomp.to_i
    count = n
    mult = n-1

    for i in 1...count
        n *= mult
        mult -= 1
    end

    puts "The factorial of #{count} is #{n}"
 end
 
 

File: 04/clojure/src/factorial.clj
;;;; factorial.clj calculates the factorial of an integer.
;;;;
;;;; Input: n, an integer.
;;;; Output: the value of n!
;;;; Author: Jonathan Winkle
;;;; Date: 3/3/20
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns factorial)                                 ; name the program

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; factorial() recursively generates the value of n!.
;;; Receive: n, an integer
;;; Output: The value of n!.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

;; direct-recursive version

(defn factorial [number]
  (if (<= number 1) 1 (* number (factorial(- number 1)))) 
)


;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; -main is a 'driver' for the factorial() function.
;;; Input: n
;;; Output: n!
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(defn -main []
 (println "To calculate the factorial of an integer")
 (print " enter the integer: ") (flush)
 (let
    [ n (int (read))] 
    (println)
    (print "The factorial of " n " is " (factorial n)"\n")
 )
)

